utils/_context/_scenarios/auto_injection.py

```diff
@@ -139,21 +128,27 @@ def _check_test_environment(self):
         assert os.getenv("DD_API_KEY_ONBOARDING") is not None, "DD_API_KEY_ONBOARDING is not set"
         assert os.getenv("DD_APP_KEY_ONBOARDING") is not None, "DD_APP_KEY_ONBOARDING is not set"
 
-    def _get_warmups(self):
-        logger.terminal.write_sep("=", "Provisioning Virtual Machines", bold=True)
-        return [self.vm_provider.stack_up]
+    def get_warmups(self):
+        warmups = super().get_warmups()
+
+        if self.is_main_worker:
+            warmups.append(lambda: logger.terminal.write_sep("=", "Provisioning Virtual Machines", bold=True))
+
+        warmups.append(self.vm_provider.stack_up)
```

Comment:
The stack_up should run only once (in the main worker)

Reply:
done